tests/aws-app/utils/saveAsJson.test.ts

```diff
@@ -0,0 +1,54 @@
+import {promises as fs} from 'fs'
+import {saveAsJson} from '@/aws-app/utils/saveAsJson'
+
+jest.mock('fs', () => ({
+	promises: {
+		writeFile: jest.fn(),
+	},
+}))
+
+describe('saveAsJson', () => {
+	beforeEach(() => {
+		jest.clearAllMocks()
+	})
+
+	it('should save the file with .cloud.json extension if no extension is provided', async () => {
+		const fileName = 'testfile'
+		const data = {key: 'value'}
+		const compressed = false
+
+		await saveAsJson(fileName, data, compressed)
+
+		expect(fs.writeFile).toHaveBeenCalledWith('testfile.cloud.json', JSON.stringify(data, null, 2))
+	})
+
+	it('should save the file with the provided extension', async () => {
+		const fileName = 'testfile.json'
+		const data = {key: 'value'}
+		const compressed = false
+
+		await saveAsJson(fileName, data, compressed)
+
+		expect(fs.writeFile).toHaveBeenCalledWith('testfile.json', JSON.stringify(data, null, 2))
+	})
+
+	it('should save the file with compressed format if compressed is true', async () => {
+		const fileName = 'testfile'
+		const data = {key: 'value'}
+		const compressed = true
+
+		await saveAsJson(fileName, data, compressed)
+
+		expect(fs.writeFile).toHaveBeenCalledWith('testfile.cloud.json', JSON.stringify(data, null, 0))
+	})
+
+	it('should save the file with pretty format if compressed is false', async () => {
+		const fileName = 'testfile'
+		const data = {key: 'value'}
+		const compressed = false
+
+		await saveAsJson(fileName, data, compressed)
+
+		expect(fs.writeFile).toHaveBeenCalledWith('testfile.cloud.json', JSON.stringify(data, null, 2))
+	})
```

Comment:
Looks the same as [in this test](https://github.com/miroapp/cloud-data-import/pull/19/files#diff-14efc1bd29d3055a661544902900dd32a339fd36729935a525f7e6467f8665f5R15)

Reply:
Thanks for the catch. just improved the semantics and decoupled the content evaluation as well. could you check?